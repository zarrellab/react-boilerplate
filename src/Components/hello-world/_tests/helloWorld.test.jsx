import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

import HelloWorld from '..'


describe('HelloWorld component', () => {
  it('should render a greeting', () => {
    expect.assertions(1)
    const component = renderer.create(<HelloWorld />)
    return expect(component.toJSON()).toMatchSnapshot()
  })

  describe('search', () => {
    it('should save a search term to state', () => {
      expect.assertions(1)
      const component = mount(<HelloWorld />)
      const value = 'r'

      component.find('input').simulate('change', { target: { value } })
      return expect(component.state().searchTerm).toBe(value)
    })
  })
})
