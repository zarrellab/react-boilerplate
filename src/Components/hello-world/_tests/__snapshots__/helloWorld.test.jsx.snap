// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld component should render a greeting 1`] = `
<div
  className="center"
  id="hello-world"
>
  <h1>
    Welcome to your react app!
  </h1>
  <h3>
    Here are some tools setup to use:
  </h3>
  <input
    className="search"
    name="search"
    onChange={[Function]}
    placeholder="search..."
    type="text"
    value=""
  />
  <ul
    className="center iconList"
    id="icon-list"
  >
    <li
      className="iconLink"
      id="icon"
    >
      <a
        href="https://facebook.github.io/react/"
        rel="noreferrer noopener"
        target="_blank"
      >
        <img
          alt="react"
          height="100"
          src="mock-source-file"
          width="100"
        />
      </a>
    </li>
    <li
      className="iconLink"
      id="icon"
    >
      <a
        href="http://babeljs.io/"
        rel="noreferrer noopener"
        target="_blank"
      >
        <img
          alt="babel"
          height="100"
          src="mock-source-file"
          width="100"
        />
      </a>
    </li>
    <li
      className="iconLink"
      id="icon"
    >
      <a
        href="https://facebook.github.io/jest/"
        rel="noreferrer noopener"
        target="_blank"
      >
        <img
          alt="jest"
          height="100"
          src="mock-source-file"
          width="100"
        />
      </a>
    </li>
    <li
      className="iconLink"
      id="icon"
    >
      <a
        href="https://webpack.github.io/docs/"
        rel="noreferrer noopener"
        target="_blank"
      >
        <img
          alt="webpack"
          height="100"
          src="mock-source-file"
          width="100"
        />
      </a>
    </li>
    <li
      className="iconLink"
      id="icon"
    >
      <a
        href="http://www.seleniumhq.org/"
        rel="noreferrer noopener"
        target="_blank"
      >
        <img
          alt="selenium"
          height="100"
          src="mock-source-file"
          width="100"
        />
      </a>
    </li>
    <li
      className="iconLink"
      id="icon"
    >
      <a
        href="https://eslint.org/"
        rel="noreferrer noopener"
        target="_blank"
      >
        <img
          alt="eslint"
          height="100"
          src="mock-source-file"
          width="100"
        />
      </a>
    </li>
  </ul>
</div>
`;
